src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static void main(String[] args) throws DukeException {
+        Scanner myObj = new Scanner(System.in);
+        int counter = 0;
+        ArrayList<Task> tasks = new ArrayList<Task>();
+
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        while (true) {
+            String input = myObj.nextLine();
+            if (input.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            } else if (input.equals("list")){
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 0; i < counter; i++) {
+                    System.out.println( Integer.toString(i + 1) + "." + tasks.get(i));
+                }
+            } else if(input.length() > 3 && input.substring(0,4).equals("mark")){
+                try {
+                    int toBeMarked = Integer.parseInt(input.substring(5));
+                    Task currentTask = tasks.get(toBeMarked - 1);
+                    currentTask.markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(currentTask);
+                } catch (StringIndexOutOfBoundsException e) {
+                    throw new DukeException("☹ OOPS!!! Please tell me which task to mark as done!");
+                }
+
```

Comment:
Are the empty lines used to differentiate between the next condition in the user input processing? This is a pretty good style as it notifies readers/reviewers of the change in logic.